src/trieNode.ts

```diff
@@ -1,20 +1,31 @@
-const rlp = require('rlp')
-const ethUtil = require('ethereumjs-util')
-const { stringToNibbles, nibblesToBuffer } = require('./util/nibbles')
-const { isTerminator, addHexPrefix, removeHexPrefix } = require('./util/hex')
+import * as rlp from 'rlp'
+import { keccak256 } from 'ethereumjs-util'
+import { stringToNibbles, nibblesToBuffer } from './util/nibbles'
+import { isTerminator, addHexPrefix, removeHexPrefix } from './util/hex'
+
+enum NodeType {
+  Leaf = 'leaf',
+  Branch = 'branch',
+  Extension = 'extention', // FIXME
+  Unknown = 'unknown', // TODO
+}
+
+export class TrieNode {
+  raw: Buffer[]
+  type: NodeType
 
-module.exports = class TrieNode {
-  constructor (type, key, value) {
+  constructor(type: any, key?: any, value?: any) {
     if (Array.isArray(type)) {
-      // parse raw node
-      this.parseNode(type)
+      this.raw = type
+      this.type = TrieNode.getNodeType(type)
     } else {
       this.type = type
       if (type === 'branch') {
         var values = key
+        // @ts-ignore
         this.raw = Array.apply(null, Array(17))
         if (values) {
-          values.forEach(function (keyVal) {
+          values.forEach(function(this: any, keyVal: any) {
```

Comment:
Can't really read this code TBH, was the missing `this` a bug before or is this a structural adoption?

Reply:
Yeah, it reads like a puzzle :) No, that's just for a typescript error, it's not in reality modifying anything.